src/main/java/uk/gov/ons/census/casesvc/service/ReceiptService.java

```diff
@@ -63,4 +125,32 @@ public void processReceipt(ResponseManagementEvent receiptEvent) {
         receiptEvent.getEvent(),
         convertObjectToJson(receiptPayload));
   }
+
+  private void logUnlinkedUacQidLink(UacQidLink uacQidLink, ResponseManagementEvent receiptEvent) {
+    log.with("qid", receiptEvent.getPayload().getResponse().getQuestionnaireId())
+            .with("tx_id", receiptEvent.getEvent().getTransactionId())
+            .with("channel", receiptEvent.getEvent().getChannel())
+            .warn("UnReceipt received for unaddressed UAC/QID pair not yet linked to a case, ");
+  }
+
+  private boolean hasCaseAlreadyBeenReceiptedByAnotherQidOrIsCaseNull(
+      UacQidLink receivedUacQidLink) {
+
+    Case caze = receivedUacQidLink.getCaze();
+
+    // If null then it's unlinked so we'll return true as we don't want to update the case, or lack
+    // thereof
+    if (caze == null) return true;
```

Comment:
Don't really need this bit. We check that the case is not null before hand. We can just pass in the case

Reply:
Fair shout, I think it got moved there, but as you say it's now impossible to call it.